oarepo_oidc_einfra/tasks.py

```diff
@@ -161,42 +208,134 @@ def update_from_perun_dump(dump_url, fix_communities_in_perun=True):
     synchronize_users_from_perun(dump, community_support)
 
 
-def synchronize_communities_to_perun(repository_community_roles, aai_community_roles):
-    resource_community_roles: List[Tuple[str, str]]
+def synchronize_communities_to_perun(
+    repository_community_roles: set[CommunityRole],
+    aai_community_roles: set[CommunityRole],
+) -> None:
+    """Synchronize communities to perun if they do not exist in perun yet.
 
+    :param repository_community_roles:   set of community roles from the repository
+    :param aai_community_roles:          set of community roles from the perun dump
+    """
     if repository_community_roles - aai_community_roles:
         log.info(
             "Some community roles are not mapped "
             f"to any resource: {repository_community_roles - aai_community_roles}"
         )
-        unsynchronized_communities = {
-            cr.community_id for cr in repository_community_roles - aai_community_roles
+        communities_not_in_perun = {
+            str(cr.community_id)
+            for cr in repository_community_roles - aai_community_roles
         }
-        for community_id in unsynchronized_communities:
-            synchronize_community_to_perun.delay(community_id)
+        for community_id in communities_not_in_perun:
+            synchronize_community_to_perun(community_id)
+
+
+def chunks[T](iterable: Iterable[T], size: int = 10) -> Iterable[chain[T]]:
+    """Split the iterable into chunks of the given size.
+
+    :param iterable:    an iterable that will be split to chunks
+    :param size:        size of the chunk
+    """
+    iterator = iter(iterable)
+    for first in iterator:
+        yield chain([first], islice(iterator, size - 1))
 
 
-def synchronize_users_from_perun(dump, community_support):
+def synchronize_users_from_perun(
+    dump: PerunDumpData, community_support: CommunitySupport
+) -> None:
+    """Synchronize users from perun dump to the database.
+
+    :param dump:                 perun dump data
+    :param community_support:    community support object
+    """
     local_users_by_einfra = einfra_to_local_users_map()
-    for aai_user in dump.users():
-        local_user_id = local_users_by_einfra.pop(aai_user.einfra_id, None)
-        # do not create new users proactively, we can do it on the first login
-        if not local_user_id:
+    print([aai_user.email for aai_user in dump.users()])
```

Comment:
Remove print to console before production

Reply:
I'd keep prints for the time being to be here before we agree on audit logs - then these prints should be replaced with writing to audit logs